[mipi_dsi] add M5Stack-Tab5-V2 model

miniskipper opened the pull request:
## Description:

Add M5Stack-Tab5-V2 model

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** 

- esphome/esphome#12074

## Checklist:

  - [x] I am merging into `next` because this is new documentation that has a matching pull-request in [esphome](https://github.com/esphome/esphome) as linked above.  
    or
  - [ ] I am merging into `current` because this is a fix, change and/or adjustment in the current documentation and is not for a new component or feature.

  - [ ] Link added in `/components/index.rst` when creating new documents for new components or cookbook.

<details>
<summary><strong>New Component Images</strong></summary>

If you are adding a new component to ESPHome, you can automatically generate a standardized black and white component name image for the documentation.

**To generate a component image:**

1. Comment on this pull request with the following command, replacing `COMPONENT_NAME` with your component name in **UPPER_CASE** format with **underscores** (e.g., `BME280`, `SHT3X`, `DALLAS_TEMP`):

   ```
   @esphomebot generate image COMPONENT_NAME
   ```

2. The ESPHome bot will respond with a downloadable ZIP file containing the SVG image.

3. Extract the SVG file and place it in the `images/` folder of this repository.

4. Use the image in your component's index table entry in `/components/index.rst`.

**Example:** For a component called "DHT22 Temperature Sensor", use:
```
@esphomebot generate image DHT22
```

</details>

miniskipper committed:
[mipi_dsi] add M5Stack-Tab5-V2 model

miniskipper committed:
Merge branch 'next' into current-mipi_dsi

miniskipper committed:
Merge branch 'next' into current-mipi_dsi

clydebarrow commented on content/components/binary_sensor/_index.md:
```diff
@@ -322,10 +322,10 @@ binary_sensor:
       - logger.log:
           format: "New state is %s"
-          args: ['x.has_value() ? ONOFF(x) : "Unknown"']
+          args: ['x.has_value() ? ONOFF(x.value()) : "Unknown"']
```
This doesn't belong here - remove it and raise a separate PR.

clydebarrow commented on content/components/display/mipi_dsi.md:
```diff
@@ -45,6 +45,7 @@ specified, or a custom init sequence can be provided.
 | JC1060P470             | Guition      | <https://aliexpress.com/item/1005008328088576.html>                           |
 | JC4880P443             | Guition      | <https://aliexpress.com/item/1005009618259341.html>                           |
 | M5STACK-TAB5           | M5Stack      | <https://shop.m5stack.com/products/m5stack-tab5-iot-development-kit-esp32-p4> |
+| M5STACK-TAB5-V2        | M5Stack      | <https://shop.m5stack.com/products/m5stack-tab5-iot-development-kit-esp32-p4> |
```
Need a note about how to distinguish the two models.

clydebarrow committed:
Merge branch 'next' into current-mipi_dsi

dcgrove committed:
Add M5Stack Tab5 model identification details

Added identification details for M5Stack Tab5 models, including display driver information and configuration guidance.

dcgrove committed:
Add files via upload

clydebarrow committed:
Merge branch 'next' into current-mipi_dsi

miniskipper committed:
Merge pull request #1 from dcgrove/current-mipi_dsi

Add Tab5 vs Tab5-V2 model identification documentation

miniskipper committed:
Merge branch 'next' into current-mipi_dsi

clydebarrow commented on content/components/display/mipi_dsi.md:
```diff
@@ -46,9 +46,17 @@ specified, or a custom init sequence can be provided.
 | WAVESHARE-P4-86-PANEL | Waveshare | <https://www.waveshare.com/esp32-p4-wifi6-touch-lcd-4b.htm?sku=31570> |
 
+{{< alert >}}
+**M5Stack Tab5 Model Identification:** The Tab5 has two hardware revisions with different display controllers. Units manufactured before October 14, 2025 use the ILI9881C display driver with separate GT911 touch driver (use `M5STACK-TAB5`). Units manufactured on or after that date use the integrated ST7123 display-touch driver (use `M5STACK-TAB5-V2`). If unsure which model you have, ESPHome will  ...
```
```suggestion
**M5Stack Tab5 Model Identification:** The Tab5 has two hardware revisions with different display controllers. Units manufactured before October 14, 2025 use the ILI9881C display driver with separate GT911 touch driver (use `M5STACK-TAB5`). Units manufactured on or after that date use the integrated ST7123 display-touch driver (use `M5STACK-TAB5-V2`). If unsure which model you have, check the sticker on the back of the device for the display driver chip name. The label is just above the ESPressif icon. See image below for example of V2 hardware.

Selection of the wrong display driver model will cause the display to simply fail to work with no relevant logging. Selection of the wrong touchscreen driver however will display an error message in the log output, so if in doubt, verify the correct touchscreen driver first to accurately identify the board before configuring the display driver.
```

clydebarrow committed:
Clarify description

clydebarrow committed:
Fix image; format text

clydebarrow committed:
whitespace

clydebarrow committed:
Add new imagte
